github.go

```diff
@@ -57,22 +77,17 @@ func mergePullRequest(owner, repo, number string) error {
 		log.Printf("Attempting to merge PR #%s on %s/%s...\n", number, owner, repo)
 	}
 
-	req, err := newMergeRequest(owner, repo, number)
-	if err != nil {
-		return err
-	}
-
-	var merged map[string]interface{}
-	_, mergeErr := req.Put(map[string]string{}, &merged)
+	commitMsg := fmt.Sprintf("Merge pull request %v", number)
+	_, _, mergeErr := client.PullRequests.Merge(owner, repo, stringToInt(number), commitMsg)
 
 	if mergeErr != nil {
 		if verbose {
 			fmt.Println("Received an error!", mergeErr)
 		}
-		if strings.Contains(mergeErr.Error(), "405 - Pull Request is not mergeable") {
+		if strings.Contains(mergeErr.Error(), "405 Pull Request is not mergeable") {
```

Comment:
you need to type assert it as a github.Error (see http://blog.golang.org/error-handling-and-go)